MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -2419,77 +2568,357 @@ private static void correctNonAeroFlyerBehavior(List<Entity> entityList, int boa
     }
 
     /**
-     * Helper function that makes some of the units in the given list of entities
-     * carry bombs.
-     * @param entityList The list of entities to process
-     * @param campaign Campaign object. In the future, may be used to check list of bombs
-     * for technological availability.
+     * Helper function to load bombs onto a random portion of units that can carry them
+     * @param entityList        The list of entities to process
+     * @param campaign          Campaign object
+     * @param hasGroundTargets  true to select air-to-ground ordnance, false for air-to-air only
+     * @param quality           IUnitRating enum for force quality (A/A* through F)
+     * @param isPirate          true to use specific pirate ordnance loadouts
      */
-    public static void populateAeroBombs(List<Entity> entityList, Campaign campaign, boolean groundMap) {
-        int maxBombers = Compute.randomInt(entityList.size()) + 1;
+    public static void populateAeroBombs(List<Entity> entityList,
+                                         Campaign campaign,
+                                         boolean hasGroundTargets,
+                                         int quality,
+                                         boolean isPirate) {
+
+        // Get all valid bomber
+        List<Entity> bomberList = entityList.stream().filter(Targetable::isBomber).collect(Collectors.toList());
+        if (bomberList.isEmpty()) {
+            return;
+        }
+
+        int minThrust;
+        int maxLoad;
+        int maxBombers = (int) Math.ceil(((Compute.randomInt(60) + 40) / 100.0) * bomberList.size());
         int numBombers = 0;
 
-        int[] validBombChoices = groundMap ? validBotBombs : validBotAABombs;
+        int[] generatedBombs = new int[BombType.B_NUM];
+        Map<Integer, int[]> bombsByCarrier = new HashMap<>();
 
-        for (Entity entity : entityList) {
-            if (entity.isBomber()) {
-                // if this entity has no guns (e.g. is a Boeing Jump Bomber)
-                if (entity.getIndividualWeaponList().isEmpty()) {
-                    loadBombs(entity, validBombChoices, campaign.getGameYear());
-                    continue;
+        boolean hasGuided = false;
+        for (int i = 0; i < bomberList.size(); i++) {
+            if (numBombers >= maxBombers) {
+                break;
+            }
+
```

Comment:
Is it worth adding a check here? I'm guessing that if the unit is unarmed it's not going to need any bombs.

```
if (isUnarmed) {
    continue
}
```

Reply:
Possible relevant: The Boeing Jump Bomber is specifically designed to drop bombs, but has no internal weapons.

Reply:
Carried this logic forward from the previous test.  If it's carrying bombs, it really should be loaded up as that's the only realistic contribution it can make.  Lower down in the code, they are specifically excluded from counting towards the total number of loaded bombers.

If anything they would be prioritized over armed units.  Would it be worth re-ordering the bomber subset List<> to put unarmed entities first so they always get loaded up?

Reply:
No, if that's the case then it's likely fine how it is. I hadn't spotted the previous test where it was defined :)